### PR TITLE
Fix issue in UIButton that we are not fallback to value of normal state when setting value of other states to be nil

### PR DESCRIPTION
**Pull request by yiyang-msft**
UIButton can set different title/color/image for different state. if the value of a state, e.g., title for highlighted state is set to nil, then when button is lightlighted state, its value should fallback to normal state value. 

Currently when we set nil to title/color/image value for a state (e.g., highlighted), we are creating a non-nil inspectable object which in turns contains null comObj, so when button is in that state (e.g., highlighted state), we use this inspectable to retrieve their contained value (null) to update UIButton, which causes button text/image display incorrectly.

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/winobjc/1493)

### CODE REVIEW

File: Frameworks/UIKit/UIButton.mm [hunk 1]
```diff
@@ -481,9 +485,19 @@ - (UIImage*)currentBackgroundImage {
 */
 - (void)setTitle:(NSString*)title forState:(UIControlState)state {
     _states[state].title.attach([title copy]);
-    RTObject* rtString = [WFPropertyValue createString:title];
-    if (rtString) {
-        _states[state].inspectableTitle = [rtString comObj];
+
+    // NOTE: check if title is nil before creating inspetableTitle
```
Comment (jaredhms) [inline Frameworks/UIKit/UIButton.mm]: >inspetableTitle [](start = 51, length = 15)

nit: typo

File: Frameworks/UIKit/UIButton.mm [hunk 2]
```diff
@@ -262,9 +262,13 @@ - (void)initAccessibility {
...
-    WUXMImageBrush* imageBrush = ConvertUIImageToWUXMImageBrush(image);
-    if (imageBrush) {
-        _states[state].inspectableImage = [imageBrush comObj];
+    if (image) {
+        WUXMImageBrush* imageBrush = ConvertUIImageToWUXMImageBrush(image);
+        if (imageBrush) {
+            _states[state].inspectableImage = [imageBrush comObj];
+        }
+    } else {
+        _states[state].inspectableImage = nil;
```
Comment (jaredhms) [inline Frameworks/UIKit/UIButton.mm]: >nil [](start = 42, length = 3)

nit: should be nullptr for all of these; you're assigning to a ComPtr.

File: Frameworks/UIKit/UIButton.mm [hunk 3]
```diff
@@ -262,9 +262,13 @@ - (void)initAccessibility {
 */
 - (void)setImage:(UIImage*)image forState:(UIControlState)state {
     _states[state].image = image;
-    WUXMImageBrush* imageBrush = ConvertUIImageToWUXMImageBrush(image);
-    if (imageBrush) {
-        _states[state].inspectableImage = [imageBrush comObj];
+    if (image) {
```
Comment (jaredhms) [inline Frameworks/UIKit/UIButton.mm]: >   if (image) { [](start = 1, length = 15)

nit: worth adding the same comment here as you did in the other two portions of the fix?

**Comment by jaredhms** [comment]
:shipit:

**Comment by tadam-msft** [comment]
Thanks

Sent from Windows Mail

**Pull request merged**